core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceRepository.scala

```diff
@@ -21,6 +32,26 @@ class WorkspaceRepository(dataSource: SlickDataSource) {
       access.workspaceQuery.findById(workspaceId.toString)
     }
 
+  def getWorkspaceContext(workspaceName: WorkspaceName,
+                          attributeSpecs: Option[WorkspaceAttributeSpecs] = None
+  ): Future[Option[Workspace]] = dataSource.inTransaction { dataAccess =>
+    dataAccess.workspaceQuery.findByName(workspaceName, attributeSpecs)
+  }
+
+  def getV2WorkspaceContext(workspaceName: WorkspaceName,
```

Comment:
Is there a use-case for retaining access to non-V2 workspaces? If not I suggest either removing the non-v2 code paths or marking them deprecated for future removal so they don't continue getting ported.

Reply:
I'm all for consolidating code paths.

I don't see much point in deprecating internal service functionality, if it's possible to just change it instead.